docs/quota_management.md

```diff
@@ -0,0 +1,102 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+---
+
+Quota Management
+===
+
+Celeborn allows the administrator to set quotas for the number of names used.
+The administrator also can set a system level default quota for all user that 
+doesn't set a specified a named quota.
+
+When Master side sets `celeborn.quota.enabled` to true, the master will enable
+to check the quota via QuotaManager. When Client side also sets `celeborn.quota.enabled`
+to true, LifecycleManager will request master side to check whether the current user identifier
+has enough quota before registration of shuffle. Fallback to the default shuffle service
+of Spark when Master side checks that there is no enough quota for current user.
+
+## Quota
+Celeborn supports fine-grained quota management, now Celeborn including four indicators:
+- `celeborn.quota.tenant.diskBytesWritten`: Maximum allowed disk write bytes, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.diskFileCount`: Maximum allowed disk write file num, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.hdfsBytesWritten`: Maximum allowed hdfs write bytes, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.hdfsFileCount`: Maximum allowed hdfs write file num, default value `Long.MAX_VALUE`.
+
+## User Identifier
+The Celeborn supports user identifier for two level:
+  - tenant id
+  - username
+The client LifecycleManager will request master side to check quota for current user that defined by user setting.
+User can set `celeborn.quota.identity.provider` to choose a identity provider, now we support two type provider:
+  - `org.apache.celeborn.common.identity.HadoopBasedIdentityProvider`: username will be obtained by `UserGroupInformation.getUserName()`, tenant id will be default.
+  - `org.apache.celeborn.common.identity.DefaultIdentityProvider`: username and tenant id are default values or user-specific values setting by `celeborn.quota.identity.user-specific.tenant` and `celeborn.quota.identity.user-specific.userName`.
+Celeborn uses `org.apache.celeborn.common.identity.DefaultIdentityProvider` as default. 
+Also, the Celeborn users can implement their own specified identify provider inheriting interface `org.apache.celeborn.common.identity.IdentityProvider` by yourself.
+
+## QuotaManager
+Celeborn will initialize a QuotaManager in the master side checking quota.
+QuotaManager uses [dynamic config service](developers/configuration.md#dynamic-configuration) to store quota settings mentioned in [Quota](#Quota).
+QuotaManger also supports two types of store backend configured by parameter `celeborn.dynamicConfig.store.backend` to store quota setting:
+  - FS: [FileSystem Store Backend](#FileSystem Store Backend)
+  - DB: [Database Store Backend](#Database Store Backend)
+
+Currently, QuotaManager supports [three level](developers/configuration.md#Config Level) quota settings.
+
+### FileSystem Store Backend
```

Comment:
We could introduce an example for quota management and provide how to config quota for this example.